Port release/gefs_v12 to WCOSS2

XianwuXue-NOAA opened the pull request.

XianwuXue-NOAA committed:
Update GSL-prep-chem & FV3-GSDCHEM-WW3 from vlab to github

 On branch feature/gefs_v12_github
 Changes to be committed:
	modified:   sorc/checkout.sh

Refs: #348

XianwuXue-NOAA committed:
Change the components in g-w to use my repo

For the next porting to acorn

 On branch feature/gefs_v12_github_port2acorn
 Changes to be committed:
	modified:   sorc/checkout.sh

Refs: #345

XianwuXue-NOAA committed:
Add acorn to sorc/machine-setup.sh

 On branch feature/gefs_v12_github_port2acorn
 Changes to be committed:
	modified:   sorc/machine-setup.sh

Refs: #345

XianwuXue-NOAA committed:
Port program enkf_chgres_recenter_nc to acorn

 On branch feature/gefs_v12_github_port2acorn
 Changes to be committed:
	new file:   modulefiles/fv3gfs/enkf_chgres_recenter_nc.acorn

Refs: #345

XianwuXue-NOAA committed:
Port gfs_bufr & tocsbufr to acorn

Issue is there is no SIGIO_LIB4 & SIGIO_INC4 in module sigio/2.3.2
	I use SIGIO_INC and SIGIO_LIB to assign values to them

 On branch feature/gefs_v12_github_port2acorn
 Changes to be committed:
	new file:   modulefiles/gfs_bufr.acorn

Refs: #345

XianwuXue-NOAA committed:
Port sorc/build_gsd_prep_chem.sh to acorn

 On branch feature/gefs_v12_github_port2acorn
	modified:   sorc/build_gsd_prep_chem.sh

Refs: #345

XianwuXue-NOAA committed:
Add acorn to build_ufs_utils.sh

 On branch feature/gefs_v12_github_port2acorn
	modified:   sorc/build_ufs_utils.sh

Refs: #345

XianwuXue-NOAA committed:
Update modules after Acorn environment changed

 On branch feature/gefs_v12_github_port2acorn
	modified:   modulefiles/fv3gfs/enkf_chgres_recenter_nc.acorn
	modified:   modulefiles/gfs_bufr.acorn
	modified:   sorc/machine-setup.sh

Refs: #345

XianwuXue-NOAA committed:
Update the checkout repo/branches for transition
 On branch feature/gefs_v12_github_port2acorn
 Changes to be committed:
	modified:   sorc/checkout.sh

Refs: #345

XianwuXue-NOAA committed:
Change acorn to wcoss2

 On branch feature/gefs_v12_github_port2acorn
	renamed:    modulefiles/fv3gfs/enkf_chgres_recenter_nc.acorn -> modulefiles/fv3gfs/enkf_chgres_recenter_nc.wcoss2
	renamed:    modulefiles/gfs_bufr.acorn -> modulefiles/gfs_bufr.wcoss2
	modified:   sorc/build_gsd_prep_chem.sh
	modified:   sorc/build_ufs_utils.sh
	modified:   sorc/machine-setup.sh

Refs: #345

XianwuXue-NOAA committed:
Remove jlogfile and modify format of wavempexec

1) Remote jlogfile
2) Modify format of wavempexec, however, may be changed
	in the future

 On branch feature/gefs_v12_github_port2acorn
 Changes to be committed:
	modified:   scripts/exwave_init.sh
	modified:   scripts/exwave_prep.sh
	modified:   ush/wave_grid_moddef.sh

Refs: #345

XianwuXue-NOAA committed:
Update the checkout.sh to checkout latest repos

which have ported to WCOSS2

 On branch feature/gefs_v12_github_port2acorn
 Changes to be committed:
	modified:   checkout.sh

Refs: #345

XianwuXue-NOAA committed:
Port link_fv3gfs.sh to WCOSS2 and improve it

To link needed parm file for chem to parm
 On branch feature/gefs_v12_github_port2acorn
 Changes to be committed:
	modified:   sorc/link_fv3gfs.sh

Refs: #345

XianwuXue-NOAA committed:
Add calc_increment_ens_ncio.x to link_fv3gfs.sh

 On branch feature/gefs_v12_github_port2acorn
 Changes to be committed:
	modified:   sorc/link_fv3gfs.sh

Refs: #345

XianwuXue-NOAA committed:
Remove the unused sorc folders in link_fv3gfs.sh

 On branch feature/gefs_v12_github_port2acorn
 Changes to be committed:
	modified:   sorc/link_fv3gfs.sh

Refs: #345

XianwuXue-NOAA committed:
Add version of PrgEnv-intel and use hdf5/netcdf

1) Add version of PrgEnv-intel
2) Replace hdf5-parallel/netcdf-hdf5parallel using hdf5/netcdf

 On branch feature/gefs_v12_github_port2acorn
 Changes to be committed:
	modified:   modulefiles/fv3gfs/enkf_chgres_recenter_nc.wcoss2
	modified:   modulefiles/gfs_bufr.wcoss2

Refs: #345

XianwuXue-NOAA committed:
Remove jlogfile and modify waveexe for wcoss2

1) Remove jlogfile based on new standard for wcoss2
2) Modify the format of wavempexec for wcoss2

 On branch feature/gefs_v12_github_port2acorn
 Changes to be committed:
	modified:   scripts/exglobal_fcst_nemsfv3gfs.sh
	modified:   scripts/exwave_init.sh
	modified:   scripts/exwave_post_sbs.sh
	modified:   scripts/exwave_prep.sh
	modified:   scripts/exwave_stat.sh
	modified:   ush/wave_ens_bull.sh
	modified:   ush/wave_ens_stat.sh
	modified:   ush/wave_grib2_sbs.sh
	modified:   ush/wave_grid_interp_sbs.sh
	modified:   ush/wave_grid_moddef.sh
	modified:   ush/wave_outp_spec.sh
	modified:   ush/wave_prnc_ice.sh
	modified:   ush/wave_tar.sh

Refs: #345

XianwuXue-NOAA committed:
Change back for scripts/exwave_stat.sh

 On branch feature/gefs_v12_github_port2acorn
 Changes to be committed:
	modified:   scripts/exwave_stat.sh

Refs: #345

XianwuXue-NOAA committed:
Change NETCDF4 to NETCDF,remove "-lhdf5_hl -lhdf5"

Due to the new NetCDF lib
1) Change NETCDF4 to NETCDF
2) Remove "-lhdf5_hl -lhdf5"
Update gfsfix path to Kate's path

 On branch feature/gefs_v12_github_port2acorn
 Changes to be committed:
	modified:   enkf_chgres_recenter_nc.fd/CMakeLists.txt
	modified:   enkf_chgres_recenter_nc.fd/src/makefile
	modified:   link_fv3gfs.sh

Refs: #345

XianwuXue-NOAA committed:
Update some modules using new versions

 On branch feature/gefs_v12_github_port2acorn
 Changes to be committed:
	modified:   ../modulefiles/fv3gfs/enkf_chgres_recenter_nc.wcoss2
	modified:   ../modulefiles/gfs_bufr.wcoss2

Refs: #345

XianwuXue-NOAA committed:
Update modules to new version in gfs_bufr

 On branch feature/gefs_v12_github_port2acorn
 Changes to be committed:
	modified:   modulefiles/gfs_bufr.wcoss2

Refs: #345

XianwuXue-NOAA committed:
Update build modules to remove hardwired ver nums

 On branch feature/gefs_v12_github_port2acorn
 Changes to be committed:
	modified:   modulefiles/fv3gfs/enkf_chgres_recenter_nc.wcoss2
	modified:   modulefiles/gfs_bufr.wcoss2

Refs: #345

XianwuXue-NOAA committed:
Change UPP from Xianwu.Xue repo to EMC repo

 On branch feature/gefs_v12_github_port2acorn
 Changes to be committed:
	modified:   sorc/checkout.sh

Refs: #345

XianwuXue-NOAA committed:
Update checkout.sh to use new gsi branch

 On branch feature/gefs_v12_github_port2acorn
	modified:   sorc/checkout.sh

Refs: #345

XianwuXue-NOAA committed:
Update gsd-prep-chem to use NOAA-GSL repo

 On branch feature/gefs_v12_github_port2acorn
 Changes to be committed:
	modified:   sorc/checkout.sh

Refs: #345

XianwuXue-NOAA committed:
Update checkout.sh to switch back to EMC repo

 On branch feature/gefs_v12_github_port2acorn
 Changes to be committed:
	modified:   checkout.sh

Refs: #345

XianwuXue-NOAA committed:
Merge branch 'feature/gefs_v12_github_port2acorn' of https://github.com/XianwuXue-NOAA/global-workflow into feature/gefs_v12_github_port2acorn

XianwuXue-NOAA committed:
Update gsi back to EMC repo in checkout.sh

 On branch feature/gefs_v12_github_port2acorn
 Changes to be committed:
	modified:   sorc/checkout.sh

Refs: #345

XianwuXue-NOAA committed:
Update checkout.sh to use tag of UPP

 On branch feature/gefs_v12_github_port2acorn
 Changes to be committed:
	modified:   sorc/checkout.sh

Refs: #345

XianwuXue-NOAA committed:
Change FV3-GSDCHEM-WW3 back to EMC repo

 On branch feature/gefs_v12_github_port2acorn
 Changes to be committed:
	modified:   sorc/checkout.sh

Refs: #345

XianwuXue-NOAA committed:
Clean up the unused statements

 On branch feature/gefs_v12_github_port2acorn
	modified:   scripts/exwave_init.sh
	modified:   scripts/exwave_post_sbs.sh
	modified:   scripts/exwave_prep.sh
	modified:   sorc/machine-setup.sh
	modified:   ush/wave_grid_moddef.sh

Refs: #345

WalterKolczynski-NOAA commented on sorc/checkout.sh:
```diff
@@ -26,11 +26,11 @@ err=0
     ((err+=$rc))
     cd fv3gfs.fd
-	git checkout gefs_v12.1.0
+	git checkout release/gefs_v12
```
This should be the new tag

XianwuXue-NOAA replied:
Yes, jut updated.

WalterKolczynski-NOAA commented on sorc/enkf_chgres_recenter_nc.fd/src/makefile:
```diff
@@ -1,6 +1,6 @@
 SHELL=  /bin/sh
 
-LIBS= $(FV3GFS_NCIO_LIB) $(BACIO_LIB4) $(W3NCO_LIB4) $(IP_LIB4) $(SP_LIB4) -L$(NETCDF)/lib -lnetcdff -lnetcdf -lhdf5_hl -lhdf5 -lz
+LIBS= $(FV3GFS_NCIO_LIB) $(BACIO_LIB4) $(W3NCO_LIB4) $(IP_LIB4) $(SP_LIB4) -L$(NETCDF)/lib -lnetcdff -lnetcdf -lz # -lhdf5_hl -lhdf5
```
Remove the commented libs

XianwuXue-NOAA replied:
No problem. I will delete them in the next commit

WalterKolczynski-NOAA commented on sorc/link_fv3gfs.sh:
```diff
@@ -38,6 +38,8 @@ elif [ $machine = "theia" ]; then
 elif [ $machine = "hera" ]; then
     FIX_DIR="/scratch1/NCEPDEV/global/glopara/fix"
+elif [ $machine = "wcoss2" ]; then
+    FIX_DIR="/lfs/h2/emc/global/noscrub/Kate.Friedman/glopara/FIX/fix_nco_gfsv15"
```
@KateFriedman-NOAA is this the final resting place for GFS v15 fix data?

KateFriedman-NOAA replied:
@WalterKolczynski-NOAA No, I need the global group account space to host these. That will be the final resting place, my copies are meant to be temporary. @arunchawla-NOAA Any update on setting up the global group account on WCOSS2? We have a bunch of stuff to host under it, like fix files for hand-off.

XianwuXue-NOAA replied:
Thanks @WalterKolczynski-NOAA and @KateFriedman-NOAA I am also waiting for the final resting place to freeze the code GEFSv12

WalterKolczynski-NOAA commented on sorc/link_fv3gfs.sh:
```diff
@@ -180,12 +183,12 @@ if [ $machine = dell -o $machine = hera ]; then
 fi
 
-for ufs_utilsexe in chgres_cube nemsio_get nemsio_read global_chgres ; do
+for ufs_utilsexe in chgres_cube ; do # nemsio_get nemsio_read global_chgres ; do
```
Remove old executables

XianwuXue-NOAA replied:
No problem. I will delete in the next commit.

WalterKolczynski-NOAA commented on sorc/link_fv3gfs.sh:
```diff
@@ -201,7 +204,7 @@ $LINK ../sorc/gsd_prep_chem.fd/workflow/emc-global/exec/prep_chem_sources_RADM_F
 cd ${pwd}/../sorc   ||   exit 8
     $SLINK gsi.fd/util/EnKF/gfs/src/calc_increment_ens.fd                                  calc_increment_ens.fd
-    $SLINK gsi.fd/util/EnKF/gfs/src/calc_increment_ens_gsdchem.fd                          calc_increment_ens_gsdchem.fd
+    #$SLINK gsi.fd/util/EnKF/gfs/src/calc_increment_ens_gsdchem.fd                          calc_increment_ens_gsdchem.fd
```
Remove

XianwuXue-NOAA replied:
done in the next commit

WalterKolczynski-NOAA commented on sorc/link_fv3gfs.sh:
```diff
@@ -217,13 +220,13 @@ cd ${pwd}/../sorc   ||   exit 8
 
-    $SLINK ufs_utils.fd/sorc/fre-nctools.fd/tools/shave.fd                                 shave.fd
-    for prog in filter_topo fregrid make_hgrid make_solo_mosaic ; do
+    #$SLINK ufs_utils.fd/sorc/fre-nctools.fd/tools/shave.fd                                 shave.fd
```
Remove

XianwuXue-NOAA replied:
same as before.

XianwuXue-NOAA committed:
Change checkout  branch to tag for FV3-GSDCHEM-WW3

 On branch feature/gefs_v12_github_port2acorn
 Changes to be committed:
	modified:   sorc/checkout.sh

Refs: #345

XianwuXue-NOAA committed:
Remove unused statements

 On branch feature/gefs_v12_github_port2acorn
 Changes to be committed:
	modified:   sorc/enkf_chgres_recenter_nc.fd/src/makefile
	modified:   sorc/link_fv3gfs.sh

Refs: #345

WalterKolczynski-NOAA reviewed (approved):
Good to merge. No tag until fix gets a final location.

XianwuXue-NOAA commented:
> Good to merge. No tag until fix gets a final location.

Agree. Once I got the final resting location of the fix folder, I will commit it.
